viewer/packages/camera-manager/src/CameraManager.ts

```diff
@@ -1,604 +1,58 @@
 /*!
- * Copyright 2021 Cognite AS
+ * Copyright 2022 Cognite AS
  */
 
-import * as THREE from 'three';
-import TWEEN from '@tweenjs/tween.js';
-import { ComboControls } from './ComboControls';
-import {
-  CameraManagerCallbackData,
-  CameraControlsOptions,
-  CameraChangeData,
-  PointerEventDelegate,
-  ControlsState
-} from './types';
-import { assertNever, EventTrigger, InputHandler, disposeOfAllEventListeners } from '@reveal/utilities';
-import range from 'lodash/range';
-export class CameraManager {
-  private readonly _events = {
-    cameraChange: new EventTrigger<CameraChangeData>()
-  };
-
-  private readonly _controls: ComboControls;
-  private readonly _camera: THREE.PerspectiveCamera;
-  private readonly _domElement: HTMLElement;
-  private readonly _inputHandler: InputHandler;
-  private readonly _raycaster = new THREE.Raycaster();
-
-  private isDisposed = false;
-
-  private readonly _modelRaycastCallback: (x: number, y: number) => Promise<CameraManagerCallbackData>;
-  private _onClick: ((event: MouseEvent) => void) | undefined = undefined;
-  private _onWheel: ((event: MouseEvent) => void) | undefined = undefined;
-
-  private static readonly DefaultAnimationDuration = 300;
-  private static readonly DefaultMinAnimationDuration = 300;
-  private static readonly DefaultMaxAnimationDuration = 1250;
-  private static readonly DefaultMinDistance = 0.8;
-  private static readonly DefaultMinZoomDistance = 0.4;
-  private static readonly DefaultMinimalTimeBetweenRaycasts = 0.08;
-  private static readonly DefaultCameraControlsOptions: Required<CameraControlsOptions> = {
-    mouseWheelAction: 'zoomPastCursor',
-    changeCameraTargetOnClick: false
-  };
-
-  private _cameraControlsOptions: Required<CameraControlsOptions> = { ...CameraManager.DefaultCameraControlsOptions };
-
-  public automaticNearFarPlane = true;
-  public automaticControlsSensitivity = true;
-
+import { CameraChangeData, CameraState } from './types';
+/**
+ * Camera manager interface.
+ * */
+export interface CameraManager {
   /**
-   * Reusable buffers used by functions in Cognite3dViewer to avoid allocations.
+   * Get THREE.PerspectiveCamera object for special cases. All manipulations of the camera should be done
+   * through other CameraManager methods.
+   * **/
```

Comment:
```suggestion
   * Returns the camera used for rendering in {@link Cognite3DViewer}. 
   * Note that the camera will not be modified directly by Reveal. Implementations must trigger the `cameraChange`-event whenever the
   * camera changes.
   */
```

It's up to implementations if they want to modify the camera directly, whats important is that Reveal wont do that and that cameraChanged is triggered when it is modified.